waku/v2/protocol/metadata/waku_metadata.go

```diff
@@ -225,14 +240,13 @@ func (wakuM *WakuMetadata) Connected(n network.Network, cc network.Conn) {
 		shard, err := wakuM.Request(wakuM.ctx, peerID)
 		if err == nil {
 			if shard == nil {
-				err = errors.New("no shard reported")
+				shouldDisconnect = false // No shard reported
```

Comment:
This doesn't seem right. 
The Request method can return nil, nil in case metadata protocol is not supported by the peer. Doesn't this mean we should disconnect from the peer because our own clusterID is non-Zero?

Reply:
Yes, this was a misunderstanding I had about the metadata protocol that I cleared today. Will update the PR